app/controllers/cart_controller.rb

```diff
@@ -22,12 +22,30 @@ def remove_item
     redirect_to '/cart'
   end
 
+  def edit_quantity
+    adjust_quantity
+  end
+
   private
 
   def require_not_admin
       render file: "/public/404" if current_admin
   end
 
+  def item
+    @item = Item.find(params[:item_id])
+  end
+
+  def adjust_quantity
+    if params[:quantity] == "add" && item.inventory > session[:cart][params[:item_id]]
```

Comment:
This takes the quantity param of an item and clicks add to increment quantity by one for each specific item but if the number of 'adds' goes beyond what the set quantity is available you get a flash message stating the item is out of stock